src/Common/src/Interop/Unix/System.Native/Interop.RealPath.cs

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        /// <summary>
+        /// Takes a path containing relative subpaths or links and returns the absolute path.
+        /// </summary>
+        /// <param name="path">The input path containing relative subpaths or links</param>
+        /// <param name="resultPath">The output path; will be null on error</param>
+        /// <returns>Returns 0 on succes and -1 on failure; on failure, resultPath will be null</returns>
+        internal static int RealPath(string path, out string resultPath)
+        {
+            byte[] buffer = new byte[Interop.Sys.MaxPath];  // MaxPath should be big enough every time
```

Comment:
If passing in NULL for the target buffer is acceptable such that the native implementation will malloc it, then you can simply return that out of the native function as a char*, define the P/Invoke to return a string, and the runtime will handle free'ing the malloc'd char*.  Then the managed buffer is unnecessary, and there's only one P/Invoke.

I'm curious about that, though, because although the Linux man page states:
```
If resolved_path is specified as NULL, then realpath() uses malloc(3) to allocate a buffer of up to PATH_MAX bytes to hold the resolved pathname, and returns a pointer to this buffer. The caller should deallocate this buffer using free(3).
```
the OS X man page states:
```
The resolved_name argument must refer to a buffer capable of storing at least PATH_MAX characters.
```